src/components/WalletModal/index.tsx

```diff
@@ -152,6 +187,27 @@ export default function WalletModal({
   const walletModalOpen = useModalIsOpen(ApplicationModal.WALLET)
   const toggleWalletModal = useToggleWalletModal()
 
+  const allTokens = useAllTokens()
+  const [tokenBalances, tokenBalancesIsLoading] = useAllTokenBalances()
+  const sortedTokens: Token[] = useMemo(() => {
+    void tokenBalancesIsLoading // creates a new array once balances load to update hooks
+    return Object.values(allTokens).sort(tokenComparator.bind(null, tokenBalances))
+  }, [tokenBalances, allTokens, tokenBalancesIsLoading])
+  const native = useNativeCurrency()
+
+  const sortedTokensWithETH: Currency[] = useMemo(() => {
+    // Use Celo ERC20 Implementation and exclude the native asset
+    if (!native) {
+      return sortedTokens
+    }
+    // Always bump the native token to the top of the list.
+    return native ? [native, ...sortedTokens.filter((t) => !t.equals(native))] : sortedTokens
```

Comment:
looks like native has to be true because otherwise you just return sorted tokens? you could also remove the check above if you wanted

Reply:
oh true, i lifted most of this code from CurrencySearch.tsx, made the simplification there as well.